pyanaconda/geoloc.py

```diff
@@ -117,22 +117,100 @@
+    """Check if geolocation can be used during this installation run.
+
+    And set the geolocation_enabled module attribute accordingly.
+
+    The result is based on current installation type - fully interactive vs
+    fully or partially automated kickstart installation and on the state of the
+    "geoloc*" boot/CLI options.
+
+    By default geolocation is not enabled during a kickstart based installation,
+    unless the geoloc_use_with_ks boot/CLI option is used or the Install Class
+    indicates geolocation should be used with kickstart.
+
+    Also the geoloc boot/CLI option can be used to make sure geolocation
+    will not be used during an installation, like this:
+
+    inst.geoloc=0
+
+    :param bool options_override: use with kickstart due to CLI/boot option override
+    :param bool install_class_override: use with kickstart due to install class override
+    """
+    use_geolocation = True
+
+    # don't use geolocation during image and directory installation
+    if flags.imageInstall or flags.dirInstall:
+        log.info("Geolocation is disabled for image or directory installation.")
+        use_geolocation = False
+    # don't use geolocation during kickstart installation unless explicitly
+    # requested by the user
+    elif flags.automatedInstall:
+        # check for use-with-kickstart overrides
+        if options_override or install_class_override:
+            use_geolocation = True
+        else:
+            # otherwise disable geolocation during a kickstart installation
+            use_geolocation = False
+
+    # and also check if geolocation was not disabled by boot or command like option
+    if not flags.cmdline.getbool('geoloc', True):
+        # flags.cmdline.getbool is used as it handles values such as
+        # 0, no, off and also nogeoloc as False
+        # and other values or geoloc not being present as True
+        use_geolocation = False
+
+    # log the end result
+    if use_geolocation:
+        if flags.automatedInstall:
+            if options_override:
+                log.info("Geolocation is enabled during kickstart installation due to use of the "
+                         "geoloc-use-with-ks option.")
+            if install_class_override:
+                log.info("Geolocation is enabled during kickstart installation due to "
+                         "install class override.")
+        else:
+            log.info("Geolocation is enabled.")
+    else:
+        if flags.imageInstall or flags.dirInstall:
+            log.info("Geolocation is disabled for image or directory installation.")
+        elif flags.automatedInstall:
+            log.info("Geolocation is disabled due to automated kickstart based installation.")
+        if not flags.cmdline.getbool('geoloc', True):
+            log.info("Geolocation is disabled by the geoloc option.")
+
+    global geolocation_enabled
+    geolocation_enabled = use_geolocation
```

Comment:
I don't think we want to have a global variable and it is even worse when it is used for the outside decisions. What about abstraction of this code to a Class with the variable inside of it. All the functions could be static to avoid requirement of holding the instance somewhere.
In any case I don't think we want to have `geolocation_enabled` global variable, IMHO it should be read only property for the outside of the geoloc.py source code.